File: Class10/demos/stacks-and-queues/stack.py
class Stack:
    
    def __init__(self):
        self.top = None


    def is_empty(self):
        return self.top == None

    def push(self, value):
        old_top = self.top
        self.top = Node(value)
        self.top.next = old_top

    def peek(self):

        if not self.top:
            raise IndexError("Empty Stack")

        return self.top.value

    def pop(self):

        if not self.top:
            raise IndexError("Empty Stack")
        
        current_top = self.top

        self.top = self.top.next

        return current_top.value


class Node:
    def __init__(self, value, next_=None):
        self.value = value
        self.next = next_

    

File: Class10/demos/stacks-and-queues/stack.cs
public class Stack<String>
{

    private Node top = null;

    public Stack()
    {
        // initialize as needed
    }

    public Null push(String value)
    {
        Node oldTop = top;
        top = new Node(value);
        top.next = oldTop;
    }

    public bool isEmpty()
    {
        return top == null;
    }

    public string peek()
    {

        // raise exception if empty

        return top.value;
    }

    public string pop()
    {

        // raise exception if empty

        Node oldTop = top;

        top = top.next;

        oldTop.next = null;

        return oldTop.value;
    }

}

public class Node
{
    public Object value;
    public Node next;
}
